source/docs/software/basic-programming/programming-leds.rst

```diff
@@ -0,0 +1,114 @@
+Programming LEDs
+================
+
+LED strips have been commonly used by teams for several years for a variety of reasons. They allow teams to debug robot functionality from the audience, provide a visual marker for their robot, and can simply add some visual appeal. WPILib has an API for controlling LEDs with their data pin connected via PWM.
+
+Instantiating the AddressableLED Object
+---------------------------------------
+
+You first create an ``AddressableLED`` object that takes the PWM port as an argument. It *must* be a PWM header. Then you set the number of LEDs located on your LED strip, with can be done with the ``setLength()`` function.
+
+.. important:: It is important to note that setting the length of the LED header is an expensive task and it's **not** recommended to run this periodically.
+
+After the length of the strip has been set, you'll have to create an ``AddressableLEDBuffer`` object that takes the number of LEDs as an input. You'll then call ``myAddressableLed.setData(myAddressableLEDBuffer)`` to set the led output data. Finally, you can call ``myAddressableLed.start`` to write the output continuously. Below is a full example of the initialization process.
+
+.. note:: C++ does not have an AddressableLEDBuffer, and instead uses an Array.
+
+.. tabs::
+
+   .. group-tab:: Java
+
+      .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/master/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/addressableled/Robot.java
+         :language: java
+         :lines: 8-23
+         :linenos:
+         :lineno-start: 14
+
+   .. group-tab:: C++
+
+      .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/master/wpilibcExamples/src/main/cpp/examples/AddressableLED/cpp/Robot.cpp
+         :language: cpp
+         :lines: 14-23
+         :linenos:
+         :lineno-start: 14
+
+      .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/master/wpilibcExamples/src/main/cpp/examples/AddressableLED/cpp/Robot.cpp
+         :language: cpp
+         :lines: 41-47
+         :linenos:
+         :lineno-start: 41
+
+Setting the Entire Strip to One Color
+-------------------------------------
+
+Color can be set to an individual led on the strip using the ``setHSV()`` method. ``setHSV`` takes four arguments; index of the LED, hue, saturation, and value.
```

Comment:
It would probably be good to have a sentance about what each of Hue, Saturation, and Value look like